### PR TITLE
Add data_driven_component taxonomy

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-engagement.atlassian.net/browse/SW-1056

### Changes
1. install `data_driven_component` vocabulary with terms
2. add a validation to ensure that only the values that has correct format are accepted.
3. 
image
image
image

### CODE REVIEW

File: tide_data_driven_component.install [hunk 1]
```diff
@@ -37,3 +38,50 @@ function tide_data_driven_component_update_8001() {
...
+    $storage = \Drupal::entityTypeManager()->getStorage($type);
+    $config_entity = $storage->createFromStorageRecord($config_read);
+    $config_entity->save();
+  }
+}
+
+/**
+ * Adding data_driven_component terms.
+ */
+function tide_data_driven_component_update_8003() {
```
Comment (anthony-malkoun) [inline tide_data_driven_component.install:68]: These values are only valid for `content-vic`?

Comment (anthony-malkoun) [reply tide_data_driven_component.install:68]: Can this hook look at current `DATA_DRIVEN_OPTIONS` and create the terms based on that?

Comment (vincent-gao) [reply tide_data_driven_component.install:68]: ! Thanks @anthony-malkoun , great point!

File: config/install/field.field.taxonomy_term.data_driven_component.field_machine_name.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+dependencies:
+  config:
+    - field.storage.taxonomy_term.field_machine_name
+    - taxonomy.vocabulary.data_driven_component
+id: taxonomy_term.data_driven_component.field_machine_name
+field_name: field_machine_name
+entity_type: taxonomy_term
+bundle: data_driven_component
+label: 'Machine Name'
+description: "example: \r\n<code>myvic_family_violence_women_programs</code> . <strong>When you entre the value, please confirm it with developers.</strong>"
```
Comment (anthony-malkoun) [inline config/install/field.field.taxonomy_term.data_driven_component.field_machine_name.yml]: *enter

File: tide_data_driven_component.install (hunk 1 above)
Comment (vincent-gao) [reply tide_data_driven_component.install:68]: Thanks @anthony-malkoun , Added https://github.com/dpc-sdp/tide_data_driven_component/pull/7/files#diff-bc568f94435aff13db5e6aa564e051f72987d50386320f959a3a79782cb9678bR90 !

File: config/install/field.field.taxonomy_term.data_driven_component.field_machine_name.yml (hunk 2 above)
Comment (vincent-gao) [reply config/install/field.field.taxonomy_term.data_driven_component.field_machine_name.yml]: changed 😬 Thanks!

File: tide_data_driven_component.install (hunk 1 above)
Comment (vincent-gao) [reply tide_data_driven_component.install:68]: > These values are only valid for content-vic?

yes, should I put the hook into vicgovau_core module? I think the answer will be yes?

Comment (anthony-malkoun) [reply tide_data_driven_component.install:68]: I think it's fine here as if `DATA_DRIVEN_OPTIONS` is empty it just won't create any terms?

Comment (vincent-gao) [reply tide_data_driven_component.install:68]: yes. 
I just removed this one https://github.com/dpc-sdp/tide_data_driven_component/pull/7/commits/443a0b79623b03574291b5c46ada30328b4714e3#diff-bc568f94435aff13db5e6aa564e051f72987d50386320f959a3a79782cb9678bL68 , doesn't really need.

**Review by anthony-malkoun — APPROVED** [review]
These changes look good Vincent!

**Review by anthony-malkoun — CHANGES_REQUESTED** [review]
Actually @vincent-gao just looked at why the builds are failing. They're all linting errors. Can you fix before merging please.

**Pull request merged**